src/tds/unittests/convert.c

```diff
@@ -321,8 +322,7 @@ main(int argc, char **argv)
 		 */
 
 		result = tds_convert(ctx, srctype, src, srclen, desttype, &cr);
-		if (result >= 0)
-			free_convert(desttype, &cr);
+		free_convert(desttype, &cr);
```

Comment:
That change I don't understand. Data should not be allocated if the result is a failure.

Reply:
Ah, yeah, this is another, evidently misplaced, fixup for the unmerged 07a8138 from #219.  The odd thing is that the `tds`-layer portions of that change don't add any fully new code, just unblock some options.